pkg/auth/oauth2authenticator.go

```diff
@@ -150,6 +158,7 @@ func NewOAuth2AuthenticatorWithCustomFuncs(
 		token:              token,
 		openBrowserFunc:    openBrowserFunc,
 		shutdownServerFunc: shutdownServerFunc,
+		tokenRefresherFunc: refreshToken,
```

Comment:
as described below, it would be good to be able to also register observer functions that are executed on tokenRefresh/tokenChange

Reply:
I'll add a PR as a proposal.

Reply:
see https://github.com/snyk/go-application-framework/pull/54